src/NUnitFramework/framework/Attributes/TestCaseSourceAttribute.cs

```diff
@@ -241,6 +246,8 @@ private IEnumerable<ITestCaseData> GetTestCasesFor(IMethodInfo method)
                             foreach (string cat in this.Category.Split(new char[] { ',' }))
                                 parms.Properties.Add(PropertyNames.Category, cat);
 
+                        TestCaseAttribute.SpecialArgumentsHandling(parms, method.GetParameters());
```

Comment:
Ah! Now I see why you refactored. Please don't make one attribute depend on another like this. Aside from having to work correctly, our attributes are supposed to show people how to write attributes and they can't depend on this sort of thing. In this case, you could create a helper class that does the job easily enough. However, I suggest you don't make any code changes till we decide whether we want this feature... see overall comments.